### PR TITLE
Remove setting `calva.paredit.strictPreventUnmatchedClosingBracket`

### LINKED ISSUE #2500: unexpected cursor movement when using 'Strict Prevent Unmatched Closing Brackets'
With this text:

```
(a {|})
```

Type `}` and expect:
```
(a {}|)
```

Actual:
```
  |(a {})
```

With `Strict Prevent Unmatched Closing Brackets` unchecked, this behaves as expected.

### PR DESCRIPTION
**Pull request by PEZ**
Remove setting `calva.paredit.strictPreventUnmatchedClosingBracket` and all handling of it.

Replace with defaults of `editor.autoClosingOvertype` `always`.

* Fixes #2500

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Made sure there is an issue registered with a clear problem statement that this PR addresses, (created the issue if it was not present).
    - [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [x] Tests
  - [x] Tested the particular change
  - [x] Figured if the change might have some side effects and tested those as well.
- [x] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [x] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe, @corasaurus-hex, @Cyrik

### CODE REVIEW

**Review by riotrah — APPROVED** [review]
The code changes seem fine, tho a lil concerned that we lose the actual intended benefit of preventing unmatched closing tokens.

I wonder if the fix for the weird indentation bug/behavior could be better fixed by a more surgical change in the post-backspace cursor realignment (ie instead of `paredit.close`, try something else)

It's possible I'm misunderstanding which function/setting does what, and that this change won't affect my desired behavior

**Comment by PEZ** [comment]
> a lil concerned that we lose the actual intended benefit of preventing unmatched closing tokens

Which actual intended benefits are they? This is a real question. I implemented (or allowed the  implementation, not recalling exactly) this feature without fully understanding what it brought to the table. I thought it was only trying to prevent accidental structure breakage at the typing of a closing bracket.

**Comment by PEZ** [comment]
I've already noticed differences in how the VS Code feature works and what we tried to do with our ugly hack. E.g. typing a closing bracket that has no opening counterpart is allowed. And also typing a bracket which is already there causes the document to get dirty, even though only the cursor moved.

Hopefully we can figure out how to do this thing properly and reintroduce it.

**Comment by riotrah** [comment]
Yeah - preventing typing unmatched closing tokens is the main intended benefit I wanted to preserve.

**Comment by riotrah** [comment]
I haven't looked tooooo hard into the implementation to really get a sense of what makes it so "hacky", nor what we could do to both allow vsc to handle the overwrite whilst handling unmatched closing tokens after that, but I suppose we could see what users say

File: docs/site/customizing.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Calva sets some VS Code settings for all Clojure files. Some of these are needed
         "editor.wordSeparators": "\t ()\"':,;~@#$%^&{}[]`",
         "editor.autoClosingBrackets": "always",
         "editor.autoClosingQuotes": "always",
+        "editor.autoClosingQuotes": "always",
```
Comment (bpringe) [inline docs/site/customizing.md:29]: @PEZ This is duplicated, but I doubt it matters, functionally.

**Pull request merged**